frontend/packages/core-auth/src/components/AuthWrapper.tsx

```diff
@@ -78,30 +92,36 @@ const AuthWrapper: React.FC<AuthWrapperProps> = ({
   useEffect(() => {
     if (!tokenResponse) return;
 
-    const token = (() => {
-      switch (injectToken) {
-        case 'access_token':
-          return tokenResponse?.accessToken ?? '';
-        case 'id_token':
-          return tokenResponse?.idToken ?? '';
-        default:
-          return '';
-      }
-    })();
-
-    onTokenChange(token);
+    switch (injectToken) {
+      case 'access_token':
+        setToken(tokenResponse?.accessToken ?? '');
+        break;
+      case 'id_token':
+        setToken(tokenResponse?.idToken ?? '');
+        break;
+      default:
+        setToken('');
+    }
   }, [tokenResponse?.accessToken, tokenResponse?.idToken]);
```

Comment:
I think this is unnecessary. I think calculating the value of `token` on each render is very inexpensive and adding it into a state variable adds too much complexity.